plotly/grid_objs/grid_objs.py

```diff
@@ -148,7 +150,21 @@ def __init__(self, columns_or_json, fid=None):
         ```
         """
         # TODO: verify that columns are actually columns
-        if isinstance(columns_or_json, dict):
+        if pd and isinstance(columns_or_json, pd.DataFrame):
+            column_names = [name for name in columns_or_json.columns]
+            duplicate_name = utils.get_first_duplicate(column_names)
```

Comment:
ah, here we go, can you not just do `utils.get_first_duplicate(columns_or_json.columns)`?